src/cart_items/cartItem.service.ts

```diff
@@ -31,6 +31,33 @@ export class CartItemService {
       .exec();
   }
 
+  async findAllByAdmin(findAllCartItem: FindAllCartItemByAdminDto): Promise<{
+    data: CartItem[];
+    total: number;
+  }> {
+    const { limit = 0, offset = 0, user_id, sort, desc } = findAllCartItem;
+
+    const optionalQuery = {};
+    if (user_id) {
+      optionalQuery["user_id"] = user_id;
+    }
+    const sorting = sort ? { [sort]: desc } : { createdAt: -1 };
+    const cartItems = await this.cartItemModel
+      .find({
+        $and: [{ isDeleted: false }],
+        $or: [optionalQuery],
+      })
+      .sort(sorting as { [key: string]: SortOrder | { $meta: "textScore" } })
```

Comment:
what's this sort mean?

Reply:
sort can determine the sort order of the returned data
the reference is https://mongoosejs.com/docs/api.html#query_Query-sort